### PR TITLE
Allow disable the back button by setting a flag in constructor

### PR DESCRIPTION
**Pull request by johanneslagos**
This will open a prompt when user press the back button. If the user choose to continue it will clear the cache too.

### CODE REVIEW

File: library/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <resources>
     <string name="app_name">Android SDK</string>
     <string name="notification_error_ssl_cert_invalid">Invalid SSL certificate. Do you want to continue?</string>
+    <string name="queue_it_exiting">You are about to exiting the queue and the app, are you sure?</string>
```
Comment (mbp) [inline library/src/main/res/values/strings.xml:4]: But is the text correct, I don't believe the app will be exited by using the back button? (Only if it is the last activity)

Suggestion: "You are about to exit the queue, are you sure?"

File: library/src/main/java/com/queue_it/androidsdk/QueueActivity.java [hunk 2]
```diff
@@ -182,8 +184,45 @@ private void broadcastQueuePassed(String queueItToken) {
...
+                        public void onClick(DialogInterface dialog, int which) {
+
+                        }
+                    })
+                    .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            broadcastQueueAborted();
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+                                finishAffinity();
```
Comment (mbp) [inline library/src/main/java/com/queue_it/androidsdk/QueueActivity.java:218]: I don't feel that we should exit the app. We should just return to the app.

Comment (johanneslagos) [reply library/src/main/java/com/queue_it/androidsdk/QueueActivity.java:218]: That is also a good idea, just return to the app and clear the cache.

**Comment by mbp** [comment]
As discussed separately, it is possible to handle the queue in a secure way without disabling the back button. Thus I am closing this.

**Pull request closed without merging**